basket/news/views.py

```diff
@@ -119,6 +122,69 @@ def ratelimited(request, e):
     }, 429)
 
 
+def generate_fxa_state():
+    return os.urandom(16).encode('hex')
+
+
+@require_safe
+def fxa_start(request):
+    fxa_oauth = get_fxa_clients()[0]
+    fxa_state = request.session['fxa_state'] = generate_fxa_state()
+    redirect_uri = request.build_absolute_uri('/fxa/success/')
+    url = fxa_oauth.get_redirect_url(state=fxa_state,
+                                     redirect_uri=redirect_uri,
+                                     scope='profile',
+                                     email=request.GET.get('email'))
+    return HttpResponseRedirect(url)
+
+
+@require_safe
+def fxa_success(request):
+    fxa_state = request.session.get('fxa_state')
+    if fxa_state:
+        # state is one-time use
+        del request.session['fxa_state']
```

Comment:
@rfk this should be fine right? I can't think of a reason to need to request it again.